### PR TITLE
Clearer code for node selection

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: pkg/overlay/cache.go [hunk 1]
```diff
@@ -92,6 +98,74 @@ func (cache *Cache) Get(ctx context.Context, nodeID storj.NodeID) (*pb.Node, err
 	return cache.db.Get(ctx, nodeID)
 }
 
+// FindStorageNodes searches the overlay network for nodes that meet the provided criteria
+func (cache *Cache) FindStorageNodes(ctx context.Context, req *pb.FindStorageNodesRequest, preferences *NodeSelectionConfig) ([]*pb.Node, error) {
+	// TODO: use a nicer struct for input
```
Comment (navillasa) [inline pkg/overlay/cache.go]: what would be a nicer struct for input? where you don't have to dig through `GetOpts().GetRestrictions()`?

File: pkg/overlay/server.go [hunk 2]
```diff
@@ -21,19 +19,19 @@ var ServerError = errs.Class("Server Error")
...
 // Server implements our overlay RPC service
 type Server struct {
-	log                 *zap.Logger
-	cache               *Cache
-	metrics             *monkit.Registry
-	nodeSelectionConfig *NodeSelectionConfig
+	log         *zap.Logger
+	cache       *Cache
+	metrics     *monkit.Registry
+	preferences *NodeSelectionConfig
```
Comment (navillasa) [inline pkg/overlay/server.go]: nit - except for newNodeRatio, aren't these more requirements than preferences?

File: satellite/satellitedb/overlaycache.go [hunk 3]
```diff
@@ -27,100 +23,53 @@ type overlaycache struct {
...
+		return nil, nil
 	}
-	defer func() {
-		err = utils.CombineErrors(err, rows.Close())
-	}()
 
-	reputableNodes, err := sqlRowsToNodes(rows)
-	if err != nil {
-		return nil, err
+	safeExcludeNodes := ""
```
Comment (navillasa) [inline satellite/satellitedb/overlaycache.go]: nit - maybe should be `safeExcludedNodes` or needs a comment to explain why it's safe

File: satellite/satellitedb/overlaycache.go [hunk 4]
```diff
@@ -27,100 +23,53 @@ type overlaycache struct {
...
-	}()
+	args = append(args, count)
 
-	newNodes, err := sqlRowsToNodes(rows)
+	rows, err := cache.db.Query(cache.db.Rebind(`SELECT node_id,
+		node_type, address, free_bandwidth, free_disk, audit_success_ratio,
+		audit_uptime_ratio, audit_count, audit_success_count, uptime_count,
+		uptime_success_count
+		FROM overlay_cache_nodes
+	`+safeQuery+safeExcludeNodes+` LIMIT ?`), args...)
```
Comment (mobyvb) [inline satellite/satellitedb/overlaycache.go]: `ORDER BY RANDOM()` right before `LIMIT`
https://github.com/storj/storj/pull/1195/files

**Pull request merged**